pkg/security/resolvers/hash/resolver_linux.go

```diff
@@ -270,30 +288,28 @@ func (resolver *Resolver) hash(eventType model.EventType, process *model.Process
 		file.HashState = model.FileOpenError
 		return
 	}
-	defer f.Close()
 
-	fi, err := f.Stat()
-	if err != nil {
-		file.HashState = model.FileNotFound
-		resolver.hashMiss[eventType][model.FileNotFound].Inc()
+	if !mode.IsRegular() {
```

Comment:
I feel like the `f == nil` check should be done before this, because we have no idea if mode is really set at this point

Reply:
We can't have `f != nil` if the file is not a regular file. moving the `f == nil` will make the `if !mode.IsRegular()` useless no ?